src/Umbraco.Infrastructure/Security/MembersIdentityUser.cs

```diff
@@ -0,0 +1,129 @@
+    /// </summary>
+    public class MembersIdentityUser : UmbracoIdentityUser
+    {
+        private string _name;
+        private string _passwordConfig;
+        private IReadOnlyCollection<IReadOnlyUserGroup> _groups;
+
+        // Custom comparer for enumerables
+        private static readonly DelegateEqualityComparer<IReadOnlyCollection<IReadOnlyUserGroup>> s_groupsComparer = new DelegateEqualityComparer<IReadOnlyCollection<IReadOnlyUserGroup>>(
+            (groups, enumerable) => groups.Select(x => x.Alias).UnsortedSequenceEqual(enumerable.Select(x => x.Alias)),
+            groups => groups.GetHashCode());
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MembersIdentityUser"/> class.
+        /// </summary>
+        public MembersIdentityUser(int userId)
+        {
+            // use the property setters - they do more than just setting a field
+            Id = UserIdToString(userId);
+        }
+
+        public MembersIdentityUser()
+        {
+        }
+
+        /// <summary>
+        ///  Used to construct a new instance without an identity
+        /// </summary>
+        public static MembersIdentityUser CreateNew(string username, string email, string memberTypeAlias, string name = null)
+        {
+            if (string.IsNullOrWhiteSpace(username))
+            {
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(username));
+            }
+
+            var user = new MembersIdentityUser();
+            user.DisableChangeTracking();
+            user.UserName = username;
+            user.Email = email;
+            user.MemberTypeAlias = memberTypeAlias;
+            user.Id = null;
+            user.HasIdentity = false;
+            user._name = name;
+            user.EnableChangeTracking();
+            return user;
+        }
+
+        /// <summary>
+        /// Gets or sets the member's real name
+        /// </summary>
+        public string Name
+        {
+            get => _name;
+            set => BeingDirty.SetPropertyValueAndDetectChanges(value, ref _name, nameof(Name));
+        }
+
+        /// <summary>
+        /// Gets or sets the password config
+        /// </summary>
+        public string PasswordConfig
+        {
+            get => _passwordConfig;
+            set => BeingDirty.SetPropertyValueAndDetectChanges(value, ref _passwordConfig, nameof(PasswordConfig));
+        }
+
+        /// <summary>
+        /// Gets or sets the user groups
+        /// </summary>
+        public IReadOnlyCollection<IReadOnlyUserGroup> Groups
+        {
+            get => _groups;
+            set
+            {
+                _groups = value.Where(x => x.Alias != null).ToArray();
+
+                var roles = new List<IdentityUserRole<string>>();
+                foreach (IdentityUserRole<string> identityUserRole in _groups.Select(x => new IdentityUserRole<string>
+                {
+                    RoleId = x.Alias,
```

Comment:
@Shazwazza Should RoleId actually be the Umbraco group ID? In which case, I'll want to change these all to accept an integrer for the role ID instead

Reply:
IIRC changing `IdentityUserRole<string>` to `IdentityUserRole<int>` is problematic and not compatible with some other generic constraints we require. `IdentityUserRole` assumes the same key type for both RoleId and UserId which isn't true if we want to use the alias which in the case of users we do (though I can't recall exactly why off the top of my head, I think identity and permission constraints generally want to use a string name for this and might also avoid an extra DB lookup). Yep for members, this would be member groups. 

If you want to try INT instead go for it, you might just have to play around with how that works or doesn't work, I just remember for users it needed to be string for one reason or another.

Reply:
Ok, thanks, will have a look at this in the next PR for roles

Reply:
FYI, I have stuck to string for roles in my PR.